dev/WebView2/WebView2.cpp

```diff
@@ -0,0 +1,1942 @@
+        appResources.TryLookup(box_value(L"SystemColorWindowColor")) :
+        appResources.TryLookup(box_value(L"SolidBackgroundFillColorBase"));
+
+    winrt::Color backgroundColor = winrt::unbox_value_or<winrt::Color>(backgroundColorAsI, winrt::Color{});
+#ifdef WINUI3
+    if (m_systemVisualBridge) { m_systemVisualBridge.BackgroundColor(backgroundColor); }
+#endif
+}
+
+HWND WebView2::EnsureTemporaryHostHwnd()
+{
+    // If we don't know the parent yet, either use the CoreWindow as the parent,
+    // or if we don't have one, create a dummy hwnd to be the temporary parent.
+    // Using a dummy parent all the time won't work, since we can't reparent the
+    // browser from a Non-ShellManaged Hwnd (dummy) to a ShellManaged one (CoreWindow).
+    winrt::CoreWindow coreWindow = winrt::CoreWindow::GetForCurrentThread();
+    if (coreWindow)
+    {
+        auto coreWindowInterop = coreWindow.as<ICoreWindowInterop>();
+        winrt::check_hresult(coreWindowInterop->get_WindowHandle(&m_tempHostHwnd));
+    }
+    else
+    {
+        // Register the window class.
+        const wchar_t CLASS_NAME[] = L"WEBVIEW2_TEMP_PARENT";
+        HINSTANCE hInstance = ::GetModuleHandleW(NULL);
+        WNDCLASSW wc = { };
+        wc.lpfnWndProc = m_fnDefWindowProcW;
+        wc.hInstance = hInstance;
+        wc.lpszClassName = CLASS_NAME;
+        m_fnRegisterClassW(&wc);
+
+        m_tempHostHwnd = m_fnCreateWindowExW(
+            0,
+            CLASS_NAME,                     // Window class
+            L"Webview2 Temporary Parent",   // Window text
+            WS_OVERLAPPED,                  // Window style
+            0, 0, 0, 0,
+            NULL,       // Parent window
+            NULL,       // Menu
+            hInstance,  // Instance handle
+            NULL        // Additional application data
+        );
+    }
+    return m_tempHostHwnd;
+}
+
+void WebView2::CreateMissingAnaheimWarning()
+{
+    auto warning = winrt::TextBlock();
+    warning.Text(L"A suitable version of Microsoft Edge WebView2 Runtime was not detected. ");
+    warning.Inlines().Append(winrt::LineBreak());
+    auto moreText = winrt::Run();
+    moreText.Text(L"Please install from: ");
+    warning.Inlines().Append(moreText);
+    auto linkText = winrt::Run();
+    linkText.Text(L"Download WebView2 Runtime");
+    auto hyperlink = winrt::Hyperlink();
+    hyperlink.Inlines().Append(linkText);
+    auto url = winrt::Uri(L"https://aka.ms/winui3/webview2download/");
+    hyperlink.NavigateUri(url);
+    warning.Inlines().Append(hyperlink);
+    Content(warning);
+}
+
+// We could have a child TextBlock (see CreateMissingAnaheimWarning), make sure it is visited by Measure pass.
+winrt::Size WebView2::MeasureOverride(winrt::Size const& availableSize)
+{
+    //if (auto child = Content().try_as<winrt::FrameworkElement>())
+    //{
+    //    Content().Measure(availableSize);
+    //}
+
```

Comment:
Do we need this for the case where WebView2 size isn't set / comes from the container, and appropriate WV2Runtime isn't present? Same for ArrangeOverride...

Reply:
No we don't need either. I can delete these.

Reply:
Ah, I guess because WebView2 is a ContentControl here... ok.